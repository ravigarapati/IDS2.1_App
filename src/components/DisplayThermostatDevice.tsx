import React, {Component} from 'react';
import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity,
  Image,
  Pressable,
} from 'react-native';
import {Typography} from '../styles';
import {Dictionary} from '../utils/dictionary';
import CustomText from './CustomText';

class DisplayThermostatDevice extends Component {
  constructor(props) {
    super(props);
  }

  state = {
    selected: false,
  };

  render() {
    return (
      <Pressable
        accessible={true}
        style={[styles.rowDirection1, styles.idsPaddingVertical]}
        onPress={() => {
          this.props.checkRadioButton(this.props.thermostat.name);
        }}>
        <View
          accessible={true}
          style={{
            width: '50%',
            alignItems: 'center',
          }}>
          <Image
            accessible={false}
            style={styles.idsImageDimensions}
            source={this.props.imagePath}
          />
        </View>
        <View style={{width: '50%'}}>
          <View
            accessible={true}
            accessibilityLabel={`${this.props.thermostat.name}. Device type: ${
              this.props.isThermostatSelected
                ? Dictionary.addDevice.thermostat
                : Dictionary.addDevice.heatPump
            }.`}
            style={[styles.columnDirection, {width: '60%', marginTop: 20}]}>
            <CustomText
              allowFontScaling={true}
              text={
                this.props.thermostat.name === 'BCC100'
                  ? this.props.thermostat.name + '\nBCC110'
                  : this.props.thermostat.name
              }
              size={14}
              font={'medium'}
              align={'left'}
            />
            <CustomText
              allowFontScaling={true}
              text={
                this.props.isThermostatSelected
                  ? Dictionary.addDevice.thermostat
                  : Dictionary.addDevice.heatPump
              }
              size={12}
              font={'medium'}
              color={'#C1C7CC'}
              align={'left'}
            />
          </View>
          <View style={{position: 'absolute', right: 5, top: true ? -5 : -20}}>
            <TouchableOpacity
              accessible={true}
              accessibilityLabel={`Radio button to select ${
                this.props.thermostat.name
              }. ${this.props.thermostat.checked ? 'Already checked' : ''}`}
              accessibilityHint={`Press it to select ${this.props.thermostat.name} as the device to add.`}
              onPress={() => {
                this.props.checkRadioButton(this.props.thermostat.name);
              }}>
              <View
                style={[
                  styles.radioButton,
                  {
                    backgroundColor: this.props.thermostat.checked
                      ? 'rgba(0, 73, 117, 1)'
                      : 'rgba(164, 171, 179, 1)',
                  },
                ]}>
                {this.props.thermostat.checked ? (
                  <View style={styles.checkedRadioButton} />
                ) : null}
              </View>
            </TouchableOpacity>
          </View>
        </View>
      </Pressable>
    );
  }
}

const styles = StyleSheet.create({
  bccPaddingVertical: {
    paddingVertical: 30,
  },
  idsPaddingVertical: {
    paddingVertical: 10,
  },
  rowDirection1: {
    flexDirection: 'row',
    borderColor: 'rgba(223,223,223,1)',
    borderWidth: 1,
    marginVertical: 5,
    paddingVertical: 30,
  },
  rowDirection: {
    flexDirection: 'row',
  },
  columnDirection: {
    flexDirection: 'column',
  },
  boldText: {
    fontWeight: 'bold',
  },
  deviceType: {
    fontSize: 12,
    color: 'gray',
  },
  radioButton: {
    height: 24,
    width: 24,
    marginLeft: 40,
    borderRadius: 12,
    alignItems: 'center',
    justifyContent: 'center',
  },
  checkedRadioButton: {
    height: 15,
    width: 15,
    borderRadius: 12,
    borderColor: 'blue',
    backgroundColor: 'white',
  },
  bccImageDimensions: {
    height: 100,
    width: 200,
    resizeMode: 'contain',
  },
  idsImageDimensions: {
    width: 140,
    height: 140,
    resizeMode: 'contain',
  },
});

export default DisplayThermostatDevice;
