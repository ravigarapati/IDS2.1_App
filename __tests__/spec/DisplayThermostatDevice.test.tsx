import React from 'react';
import renderer from 'react-test-renderer';
import DisplayThermostatDevice from '../../src/components/DisplayThermostatDevice';

describe('Thermostat display component', () => {
  test('Component to be defined', () => {
    const tree = renderer
      .create(
        <DisplayThermostatDevice
          checkRadioButton={() => {}}
          thermostat={{name: 'Test', checked: false}}
          imagePath={require('./../../src/assets/images/BCC100.png')}
          isThermostatSelected={true}
        />,
      )
      .toJSON();
    expect(tree).toBeDefined();
  });
  test('Component to be rendered', () => {
    const tree = renderer
      .create(
        <DisplayThermostatDevice
          checkRadioButton={() => {}}
          thermostat={{name: 'Test', checked: false}}
          imagePath={require('./../../src/assets/images/BCC100.png')}
          isThermostatSelected={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
