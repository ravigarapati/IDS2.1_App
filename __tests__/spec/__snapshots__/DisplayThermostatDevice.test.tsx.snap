// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thermostat display component Component to be rendered 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "borderColor": "rgba(223,223,223,1)",
        "borderWidth": 1,
        "flexDirection": "row",
        "marginVertical": 5,
        "paddingVertical": 30,
      },
      Object {
        "paddingVertical": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    style={
      Object {
        "alignItems": "center",
        "width": "50%",
      }
    }
  >
    <Image
      accessible={false}
      source={
        Object {
          "testUri": "../../../src/assets/images/BCC100.png",
        }
      }
      style={
        Object {
          "height": 140,
          "resizeMode": "contain",
          "width": 140,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "50%",
      }
    }
  >
    <View
      accessibilityLabel="Test. Device type: Thermostat."
      accessible={true}
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "marginTop": 20,
            "width": "60%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#000000",
            },
            Object {
              "fontSize": 14,
              "lineHeight": 21,
            },
            Object {
              "fontFamily": "BoschSans-Medium",
            },
            Object {},
          ]
        }
      >
        Test
      </Text>
      <Text
        accessible={true}
        numberOfLines={0}
        style={
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "#C1C7CC",
            },
            Object {
              "fontSize": 12,
              "lineHeight": 18,
            },
            Object {
              "fontFamily": "BoschSans-Medium",
            },
            Object {},
          ]
        }
      >
        Thermostat
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 5,
          "top": -5,
        }
      }
    >
      <View
        accessibilityHint="Press it to select Test as the device to add."
        accessibilityLabel="Radio button to select Test. "
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "borderRadius": 12,
                "height": 24,
                "justifyContent": "center",
                "marginLeft": 40,
                "width": 24,
              },
              Object {
                "backgroundColor": "rgba(164, 171, 179, 1)",
              },
            ]
          }
        />
      </View>
    </View>
  </View>
</View>
`;
